### PR TITLE
Replaces concatenations with template literals in plugins/spf.js

### PR DESCRIPTION
**Pull request by apoorv-mishra**
updates #2129 

Changes proposed in this pull request:
- This PR replaces string concatenations with template literals in plugins/spf.js

Checklist:
- [x] docs updated
- [x] tests updated
- [ ] [Changes](https://github.com/haraka/Haraka/blob/master/Changes.md) updated

### CODE REVIEW

File: plugins/spf.js [hunk 1]
```diff
@@ -274,15 +273,6 @@ exports.save_to_header = function (connection, spf, result, mfrom, host, id, ip)
...
-        ' (' + plugin.config.get('me') + ': domain of ' + host +
-        ((result === spf.SPF_PASS) ? ' designates ' : ' does not designate ') +
-        connection.remote.ip + ' as permitted sender) ' + [
-            'receiver=' + plugin.config.get('me'),
-            'identity=' + id,
-            'client-ip=' + (ip ? ip : connection.remote.ip),
-            'helo=' + connection.hello.host,
-            'envelope-from=<' + mfrom + '>'
-        ].join('; ')
+        `${spf.result(result)} (${plugin.config.get('me')}: domain of ${host}${result === spf.SPF_PASS ? ' designates ' : ' does not designate '}${connection.remote.ip} as permitted sender) ${[`receiver=${plugin.config.get('me')}`, `identity={id}`, `client-ip=${ip ? ip : connection.remote.ip}`, `helo=${connection.hello.host}`, `envelope-from=<${mfrom}>`].join('; ')}`
```
Comment (msimerson) [inline plugins/spf.js]: With a string literal, we can get rid of the join('; ') here too?

File: plugins/spf.js [hunk 2]
```diff
@@ -97,8 +97,7 @@ exports.hook_helo = exports.hook_ehlo = function (next, connection, helo) {
             return next();
         }
         const host = connection.hello.host;
-        plugin.log_result(connection, 'helo', host, 'postmaster@' +
-            host, spf.result(result));
+        plugin.log_result(connection, 'helo', host, `postmaster@${host, spf.result(result)}`);
```
[the remark this answers is not in the document]
Comment (apoorv-mishra) [reply plugins/spf.js]: OOPS, missed it.

File: plugins/spf.js (hunk 1 above)
Comment (apoorv-mishra) [reply plugins/spf.js]: Sure, we can. I'll make the changes and commit.

Comment (apoorv-mishra) [reply plugins/spf.js]: In fact, I missed `$` too here in  `identity=${id}`.

File: plugins/spf.js [hunk 3]
```diff
@@ -222,17 +221,17 @@ exports.hook_mail = function (next, connection, params) {
...
-        'ip=' + (ip ? ip : connection.remote.ip),
-        'domain="' + host + '"',
-        'mfrom=<' + mfrom + '>',
-        'result=' + result
+        `identity=${scope}`,
+        `ip=${ip ? ip : connection.remote.ip}`,
+        `domain="${host}"`,
+        `mfrom=<${mfrom}>`,
+        `result=${result}`
     ].join(' '));
```
Comment (msimerson) [inline plugins/spf.js]: this join is also doing string concatenation. Which means this block could be collapsed to:
```js
connection.loginfo(this, `identity=${scope} ip=${ip ? ip : connection.remote.ip} domain="${host}" mfrom=<${mfrom}> result=${result}`
```

as well. Or to make for shorter lines, the ternaries could be extracted:

```js
const show_ip=ip ? ip : connection.remote.ip
connection.loginfo(this, `identity=${scope} ip=${show_ip} domain="${host}" mfrom=<${mfrom}> result=${result}`
```

Comment (apoorv-mishra) [reply plugins/spf.js]: Sure. I'll make the changes and update PR.

**Pull request merged**